src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -156,4 +152,130 @@ public static String getCollection(Firestore database, String collectionToGetFro
+   * @throws FirebaseException
+   */
+  public static String getFirstBatch(CollectionReference collectionReference, int batchSizeLimit, String parameterToSortBy) throws FirebaseException {
+    // We have the option here to sort the collection by specific parameter beforehand (great for supporting sort-type queries later on).
+    // Setup the new Query.
+    Query page;
+    if (parameterToSortBy.equals(ServletConstantValues.DEFAULT_VALUE)) {
+      page = collectionReference.limit(batchSizeLimit);
+    } else {
+      page = collectionReference.orderBy(parameterToSortBy).limit(batchSizeLimit);
+    }
+    return getCollectionQuery(page);
+  }
+
+
+  /**
+   * Helper method for getCollectionBatch().
+   * Retrieve the next batch in the collection collectionReference of size batchSizeLimit.
+   * @param database - The database to retrieve from.
+   * @param collectionReference - A reference to the collection I want to retrieve from.
+   * @param batchSizeLimit - The maximum size of the batch, which is the size of a new page of data in infinite scroll.
+   * @param lastDocID - The ID of the last object I sent from the most recent batch.
+   * @param parameterToSortBy - The parameter to sort the list of documents by.
+   * @return - The json string representing the next batch in a request for all of the documents in a collection.
+   * @throws FirebaseException
+   */
+  public static String getNextBatch(Firestore database, CollectionReference collectionReference, int batchSizeLimit, String lastDocID, String parameterToSortBy) throws FirebaseException {
+    // First retrieve the last document I sent.
+    Query page;
+    DocumentReference documentReference = collectionReference.document(lastDocID);
+
+    // Get a "Future" for the document, which will be used to generate a DocumentSnapshot of the data point.
+    ApiFuture<DocumentSnapshot> snapshotFuture = documentReference.get();
+    DocumentSnapshot document;
+    try {
+      // Retrieve a document snapshot of the last data point.
+      document = snapshotFuture.get();
+    } catch (Exception e) {
+      // Throws a FirebaseException if we can't connect to firestore.
+      throw new FirebaseException(ServletConstantValues.UNABLE_TO_READ_FROM_FIRESTORE, e);
+    }
+
+    // If the last document I sent still exists, then retrieve a batch of size=batchSizeLimit documents 
+    //   that occur after the last doc in the collection.
+    if (document.exists()) {
+      // We have the option here to sort the collection by specific parameter beforehand (great for supporting sort queries later on).
+      if (parameterToSortBy.equals(ServletConstantValues.DEFAULT_VALUE)) {
+        page = collectionReference.limit(batchSizeLimit).startAfter(document);
+      } else {
+        page = collectionReference.orderBy(parameterToSortBy).limit(batchSizeLimit).startAfter(snapshotFuture);
+      }
```

Comment:
Here why do you use different input in `startAfter()`?

Reply:
Oof that would've been a major bug, thanks for catching that!